### PR TITLE
chore: Fix liquidation migration script for coordinator leverage = 1

### PR DESCRIPTION
**Pull request by holzeis**
follow up to review remarks on https://github.com/get10101/10101/pull/2354

### CODE REVIEW

File: coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql [hunk 1]
```diff
@@ -2,4 +2,5 @@ ALTER TABLE "positions"
     ADD COLUMN coordinator_liquidation_price REAL NOT NULL DEFAULT 0;
 
 UPDATE positions SET coordinator_liquidation_price = average_entry_price * coordinator_leverage / (coordinator_leverage + 1) where trader_direction='short';
-UPDATE positions SET coordinator_liquidation_price = average_entry_price * coordinator_leverage / (coordinator_leverage - 1) where trader_direction='long';
+UPDATE positions SET coordinator_liquidation_price = average_entry_price * coordinator_leverage / (coordinator_leverage - 1) where trader_direction='long' and coordinator_leverage > 1;
+UPDATE positions SET coordinator_liquidation_price = average_entry_price * coordinator_leverage where trader_direction='long' and coordinator_leverage = 1;
```
Comment (luckysori) [inline coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: 🤔 The formula doesn't look right to me. Elsewhere we've set the liquidation price of a party going short at 1x leverage to the maximum price the oracle can attest to. I'm not sure if that's correct though cc @bonomat.

**Review by luckysori — COMMENTED** [review]
Not sure about the formula.

File: coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql (hunk 1 above)
Comment (bonomat) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: Yea, if a party `goes` short with a leverage of 1 he basically never gets liquidated. So we can set this to a very high number. 

On the contrary, if a party goes `long` with a leverage of 1 his liquidation price is at about 50%, i.e. 
`(average_entry_price*leverage)/((leverage+1-(maintenance_margin*leverage)))` --> `average_entry_price/2`

Comment (holzeis) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: Can I get an approval then? 🙂

Comment (luckysori) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: It still looks wrong to me though. AFAICT, you're setting the coordinator's liquidation price to

```
average_entry_price * 1
```

if the coordinator goes short.

Comment (bonomat) [inline coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: Imho, based on this formula: 

```
(price_opening*leverage)/((leverage+1-(maintenance_margin*leverage)))
```

it should look like this: 

```suggestion
UPDATE positions SET coordinator_liquidation_price = average_entry_price * coordinator_leverage / (coordinator_leverage + 1) where trader_direction='long';

```

Comment (holzeis) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: whoops I've overlooked the direction.. we need the short direction as well 🙈

Comment (bonomat) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: ah, things are so confusing... don't mind me then 🙈

Comment (bonomat) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: Here we go, this should fix it: 

```suggestion
UPDATE positions SET coordinator_liquidation_price = 21_000_000 where trader_direction='long' and coordinator_leverage = 1;
```

Comment (holzeis) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql:6]: fair, I will adapt that.

Comment (holzeis) [reply coordinator/migrations/2024-04-03-105900_add_coordinator_liquidation_price/up.sql]: @bonomat I used the const from the code.

```
pub const BTCUSD_MAX_PRICE: u64 = 1_048_575;
```

**Pull request merged**